### ISSUE TITLE
Similarity-Distance-Magnitude calibration

### ISSUE THREAD

**Issue by allenschmaltz** [IINemo/lm-polygraph#340]
I recommend evaluating the estimators of predictive uncertainty as they will be used in practice, which is essentially always a selective classification setting in practice with LLMs. There will tend to not be much variation in estimators on near-in-domain data (after all, for in-domain, exchangeable/i.i.d. data we could always just learn a rescaling over the empirical CDFs), so it is critical to evaluate over distribution- and label- shifted eval sets. This latter setting is why such estimators are critical for LLM settings, since it can be difficult to anticipate such shifts over high-dimensional inputs without model assistance. Ideally, we also have some mechanism to examine the reference classes (as for interpretability-by-example of the calibration process), to control for the effective sample sizes, and to estimate the reliability of the calibration process itself. Further, we seek these properties while minimizing distributional assumptions.

My works from 2022, as well as the (now deprecated) open-source, no-code 'Reexpress one' on-device LLM-driven data analysis program (released in November 2023), introduce SOTA approaches (as of the dates they were released) for achieving these properties. The evolution of that line of work, and the recommended approach going forward, incorporating uncertainty-aware verification and interpretability-by-exemplar as intrinsic properties of the LLMs, themselves (i.e., for both train- and test-time search/compute/generation) is described in the following paper: "Similarity-Distance-Magnitude Universal Verification" (https://arxiv.org/abs/2502.20167).

**Comment by IINemo** [comment]
Thank you for your comment, in most cases we use unsupervised UQ methods and try to quantify uncertainty of general-purpose LLMs (such as LLama 3.1, GPT-4o, etc.). In this scenario it is difficult to determine what OOD instances are, since we have no dedicated training set on which we fine-tune the model.
